tests/verify_collection.py

```diff
@@ -308,15 +321,25 @@ def test_spatial_subset(collection_concept_id, env, granule_json, collection_var
     group = None
     # Try to read group in file
     with netCDF4.Dataset(subsetted_filepath) as f:
-        for g in f.groups:
-            ds = xarray.open_dataset(subsetted_filepath, group=g)
-            if len(ds.variables):
-                group = g
-                subsetted_ds = ds
-            else:
-                ds.close()
-
-    lat_var_name, lon_var_name = get_lat_lon_var_names(subsetted_ds, collection_variables)
+        ds = xarray.open_dataset(subsetted_filepath, group='')
+        lat_var_name = None
+        if len(ds.variables):
+            subsetted_ds = ds
+            lat_var_name, lon_var_name = get_lat_lon_var_names(subsetted_ds, collection_variables)
+        if lat_var_name:
+            ds.close()
+            pass
+        else:
+            for g in f.groups:
+                ds = xarray.open_dataset(subsetted_filepath, group=g)
+                if len(ds.variables):
+                    group = g
+                    subsetted_ds = ds
+                    lat_var_name, lon_var_name = get_lat_lon_var_names(subsetted_ds, collection_variables)
+
+                else:
+                    ds.close()
+
```

Comment:
I'm trying to understand the logic here. Is the idea that we should be looping through each group in the dataset trying to find lat/lon in each group and then stop when we find a group that has lat/lon variables?
Feels like it could be simplified a little bit, as it is there's a lot of if/else branching that is hard to follow.

```suggestion
        # Find the first group that has both latitude and longitude variables
        lat_var_name, lon_var_name = None, None
        for g in f.groups:
            # Open group as dataset
            ds_group = xarray.open_dataset(subsetted_filepath, group=g)
            # If group has no variables, skip it
            if len(ds_group.variables) == 0:
                ds_group.close()
                continue
            
            lat_var_name, lon_var_name = get_lat_lon_var_names(ds_group, collection_variables)
            
            if lat_var_name and lon_var_name:
                # Found lat/lon, stop looking and leave group open for reading
                group = g
                subsetted_ds = ds_group
                break
            else:
                # Group does not contain lat/lon. Close group and try next one
                ds.close()
```

Reply:
> I'm trying to understand the logic here. Is the idea that we should be looping through each group in the dataset trying to find lat/lon in each group and then stop when we find a group that has lat/lon variables? Feels like it could be simplified a little bit, as it is there's a lot of if/else branching that is hard to follow.

Yeah, I'm trying to keep going into the file if it's not found in the root group - and then to stop when it finds the lat and lon variables. Currently in the process of restructuring to make simpler - I'm open to ideas/suggestions